src/Lachain.Core/Blockchain/Pool/TransactionPool.cs

```diff
@@ -125,7 +126,7 @@ public OperatingError Add(Transaction transaction, Signature signature, bool not
             var acceptedTx = new TransactionReceipt
             {
                 Transaction = transaction,
-                Hash = transaction.FullHash(signature),
+                Hash = transaction.FullHash(signature, HardforkHeights.IsHardfork_6Active(_blockManager.GetHeight())),
```

Comment:
There is some tricky issue I think. In the blockManager, we are using the block height to check if the hardfork is active. But in this case, it's less than.
For example, let's say we set hardfork_6 = 100. 
Then, block 100 is signed with new chainId. 
But in the pool, when the current height is 99, the transactions are signed with old chanId and they will be added in block 100, those transactions will fail.

Reply:
Fixed